kratos/linear_solvers/skyline_lu_factorization_solver.h

```diff
@@ -573,211 +503,122 @@ class SkylineLUFactorizationSolver
 
     typedef typename TDenseSpaceType::MatrixType DenseMatrixType;
 
-    /// Default constructor.
     SkylineLUFactorizationSolver() {}
+
     SkylineLUFactorizationSolver(Parameters settings): BaseType(settings) {}
-    
-    /// Destructor.
-    ~SkylineLUFactorizationSolver() override {}
 
+    ~SkylineLUFactorizationSolver() override {}
 
-    /** Normal solve method.
-    Solves the linear system Ax=b and puts the result on SystemVector& rX.
-    rX is also th initial guess for iterative methods.
-    @param rA. System matrix
-    @param rX. Solution vector.
-    @param rB. Right hand side vector.
-    */
-    bool Solve(SparseMatrixType& rA, VectorType& rX, VectorType& rB) override
+    /// @copydoc BaseType::InitializeSolutionStep
+    void InitializeSolutionStep(SparseMatrixType& rLhs,
+                                VectorType& rSolution,
+                                VectorType& rRhs) override
     {
-        if(this->IsNotConsistent(rA, rX, rB))
-            return false;
-
-        const int size = TSparseSpaceType::Size(rX);
-
-        // define an object to store skyline matrix and factorization
-        LUSkylineFactorization<TSparseSpaceType, TDenseSpaceType> myFactorization;
-
-        // copy myMatrix into skyline format
-        myFactorization.copyFromCSRMatrix(rA);
-
-        // factorize it
-        myFactorization.factorize();
-
-        // and back solve
-        myFactorization.backForwardSolve(size, rB, rX);
+        KRATOS_ERROR_IF_NOT(this->IsConsistent(rLhs, rSolution, rRhs))
+            << "Inconsistent linear system: "
+            << "(" << TSparseSpaceType::Size1(rLhs) << "x" << TSparseSpaceType::Size2(rLhs) << ")"
+            << " @ "
+            << "(" << TSparseSpaceType::Size(rSolution) << "x1) "
+            << "(" << TSparseSpaceType::Size(rRhs) << "x1)";
+
+        KRATOS_TRY
+        mFactorization.clear();
+        mFactorization.copyFromCSRMatrix(rLhs);
+        mFactorization.factorize();
+        KRATOS_CATCH("")
```

Comment:
I am wondering when is this `InitializeSolutionStep` is called? Because, in the case if this is called at the normal `InitializeSolutionStep`, then we cannot support multiple static linear load cases where you only have to factorization once for all the right hand sides. Should we consider this in this PR?